src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassReader.java

```diff
@@ -2259,6 +2264,8 @@ MethodSymbol readMethod() {
         validateMethodType(name, m.type);
         setParameters(m, type);
 
+        if (Integer.bitCount(rawFlags & (PUBLIC | PRIVATE | PROTECTED)) > 1)
+            throw badClassFile("illegal.flag.combo", String.format("0x%04x", (int)rawFlags), "method", m);
```

Comment:
instead of: `String.format("0x%04x", (int)rawFlags)`, you could use method `com.sun.tools.javac.code.Flags::toString` which prints a string representation of the flags passed to it

Reply:
Great idea - thanks.